app/views/applications/show.html.erb

```diff
@@ -10,5 +22,8 @@
   <%= @application.email %>
 </p>
 
+<p>
+<%= link_to 'Delete this application', event_application_path(@event.id, @application.id), method: :delete, data: { confirm: "You sure?" } %>
```

Comment:
Maybe a more thorough confirmation message, like "Are you sure you want to delete this application?"